flax/linen/module.py

```diff
@@ -790,9 +791,52 @@ def make_rng(self, name: str) -> PRNGKey:
       raise ValueError("Can't use RNGs on unbound modules")
     return self.scope.make_rng(name)
 
+  def bind(self, variables: VariableDict, *args, rngs: RNGSequences = None,
+           mutable: CollectionFilter = False):
+    """Creates an interactive Module instance by binding variables and RNGs.
+
+    bind provides an "interactive" instance of a Module directly without
+    transforming a function with ``apply``. This is particulary useful for debugging
+    and interactive use cases like notebooks where a function would limit the ability
+    split up code into different cells.
+
+    Once the variables (and optionally RNGs) are bound to a ``Module`` it becomes a
+    stateful object. Note that idiomatic JAX is functional and therefore an interactive
+    instance does not mix well well with vanilla JAX APIs. Therefore, we recommend using
+    ``apply`` when code should be reusable and compatible across the JAX software ecosystem.
+
+    Example::
+
+      class AutoEncoder(nn.Module):
+        def setup(self):
+          self.encoder = nn.Dense(3)
+          self.decoder = nn.Dense(5)
+      
+      ae = AutoEncoder()
+      model = ae.bind(variables)
+      z = model.encode(x)
+      x_reconstructed = model.decode(z)
+
+
+    Args:
+       variables: A dictionary containing variables keyed by variable
+        collections. See :mod:`flax.core.variables` for more details
+        about variables.
+      rngs: a dict of PRNGKeys to initialize the PRNG sequences.
+        The "params" PRNG sequence is used to initialize parameters.
```

Comment:
Doesn't `variables` require initialized `"params"`?

(Maybe mention that these `rngs` will be mutated in the attribute `.scope.rngs`)